src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -640,127 +465,54 @@ private void VerifyCanCreateOperationResponseMessage(bool synchronousCall)
             this.ValidateWriterReady();
             this.VerifyCallAllowed(synchronousCall);
 
-            if (!this.rawOutputContext.WritingResponse)
+            if (!this.outputContext.WritingResponse)
             {
                 this.ThrowODataException(Strings.ODataBatchWriter_CannotCreateResponseOperationWhenWritingRequest);
             }
         }
 
         /// <summary>
-        /// Creates an <see cref="ODataBatchOperationResponseMessage"/> for writing an operation of a batch response - implementation of the actual functionality.
+        /// Creates an <see cref="ODataBatchOperationResponseMessage"/> for writing an operation of a batch response.
         /// </summary>
         /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
         /// <returns>The message that can be used to write the response operation.</returns>
-        private ODataBatchOperationResponseMessage CreateOperationResponseMessageImplementation(string contentId)
-        {
-            this.WritePendingMessageData(true);

```

Comment:
This method should remain private, contain common/state management logic, and call abstract method in derived class.